### PR TITLE
WIP: Terminology update

### PR DESCRIPTION
**Pull request by mmccool**
A PR with various terminology updates.
Also updated/added the reference to the WoT Discovery working draft and certain other terms, like TDD.

***
<a href="https://pr-preview.s3.amazonaws.com/mmccool/wot-architecture/pull/582.html" title="Last updated on Mar 4, 2021, 1:01 PM UTC (31e933d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-architecture/582/c0c8113...mmccool:31e933d.html" title="Last updated on Mar 4, 2021, 1:01 PM UTC (31e933d)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -590,11 +663,18 @@ <h1>Terminology</h1>
         Thing Description, whereas a virtual entity is the
         composition of one or more Things.</dd>
       <dt>
-        <dfn>Thing Directory</dfn>
+        <dfn>Thing Description Directory</dfn>
```
Comment (mlagally) [inline index.html:666]: "TD Directory" is used above - please align.
Is it a "Thing Description" directory or a "Thing Directory"?

File: index.html [hunk 2]
```diff
@@ -603,29 +683,30 @@ <h1>Terminology</h1>
...
-      <dd>
-        A <a>Partial TD</a> is an object that follows the same hierarchical structure of the <a>TD</a> information model, 
-        but it is not required to contain all the mandatory elements. 
-        <p class=note>
-          An example usage of a <a>Partial TD</a> is in <a>WoT Scripting API</a> as an input for the creation of <a>Exposed Things</a>. 
-          See the <a href="https://www.w3.org/TR/2020/NOTE-wot-scripting-api-20201124/#the-produce-method">produce</a> method for more details.
-        </p></dd>
-      <dt>
-        <dfn>TD Fragment</dfn>
+        <dfn>TD Element</dfn>
```
Comment (mlagally) [inline index.html:686]: We had a long discussion on these terms before we integrated them.
Please don't remove them in this MR.
If you think we need another definition, please refactor this MR.

File: index.html [hunk 3]
```diff
@@ -565,6 +618,18 @@ <h1>Terminology</h1>
         blocks. A Servient can host and expose Things and/or host Consumers that consume Things.
         Servients can support multiple Protocol Bindings to enable
         interaction with different IoT platforms.</dd>
+      <dt>
```
Comment (mlagally) [inline index.html:621]: please refactor this definition into a dedicated MR

**Comment by mlagally** [comment]
vF2F feedback:
Please refactor into smaller PRs that can be easier reviewed and don't block the entire MR if there's contentious discussion on one one of them.

**Comment by mmccool** [comment]
Closing without merging, but will refactor into smaller PRs.

**Pull request closed without merging**